empirical-org/Empirical-Core
# Adgr report new diagnostics

## WHAT
- Add new diagnostic IDs to BigQuery materialized views used in reporting Admin Diagnostics
- Tweak the "Diagnostic" selection drop-downs on the "Performance by Skill" and "Performance by Student" tabs so that it (invisibly to the user) populates the drop-downs with the appropriate Diagnostic ID values based on the currently-selected Timeframe filter
## WHY
- We want the new diagnostics that Curriculum is launching to be included in the data being reported on
- We want the complexity of the diagnostic switchover to be effectively invisible to admins, and for everything to work seamlessly.
## HOW
- Add the new diagnostic IDs to the root admin diagnostic query
- Update the underlying materialized view definition so that BigQuery includes the new diagnostics in its data rollups
- Update the logic in the code that sets Diagnostic drop-down selections to change the option values based on the the selected timeframe

### Screenshots
![Screenshot 2024-07-22 at 3 34 25 PM](https://github.com/user-attachments/assets/99b4ad39-e3d0-4b00-87e5-3ea356fff41d)

### Notion Card Links
https://www.notion.so/quill/New-Diagnostic-Engineering-Tasks-ef849d3f473a4a11a675081a89556a75?pvs=4#18c3f8030c5e4269bc80b49090bb2af6

### What have you done to QA this feature?
BACKEND
- Used a "STAGING" version of the materialized view to test functionality without impacting production (note that this code is not visible in "Files changed" because once functionality was confirmed, those changes were moved into the production materialized view definition)
- - Created a copy of the `pre_post_diagnostic_skill_group_performance_view.sql`
- - Modified it with the new logic
- - Updated queries to point at the STAGING materialized view in the branch
- - Tested all functionality in staging
FRONTEND
- Select "This school year" as the timeframe
- Select Skills tab.  Confirm that the `diagnostic_id` requested for the report was was one of the new IDs
- Select a different diagnostic from the drop-down, confirm that the `diagnostic_id` requested for the report is one of the new ones
- Repeat for Student tab.
- Select "Last school year" as the timeframe and repeat the earlier steps, but confirm that the `diagnostic_id` requested for reports are old ones
- On the Skill tab, with "This school year" selected in filters, select a non-default Diagnostic from the drop-down (Intermediate Pre), confirm that a new ID is submitted for reporting
- - Change the filter to select "Last school year"
- - Confirm that the selected Diagnostic does not change
- - Confirm that the updated report requested by the timeframe filter looks for the old ID rather than the new one
- Repeat steps for the Student tab

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | Minor tweak to keep existing tests passing, but no new tests
Have you deployed to Staging? | Yes
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
